aspnetcore/security/authentication/aad-ui.md

```diff
@@ -0,0 +1,181 @@
+---
+title: Azure Active Directory with ASP.NET Core
+author: rick-anderson
+description: Learn how to use Azure Active Directory in an ASP.NET Core app.
+ms.author: jacalvar
+ms.date: 09/22/2018
+ms.custom: "mvc, seodec18"
+no-loc: [Home, Privacy, Kestrel, appsettings.json, "ASP.NET Core Identity", cookie, Cookie, Blazor, "Blazor Server", "Blazor WebAssembly", "Identity", "Let's Encrypt", Razor, SignalR]
+uid: security/authentication/aad-ui
+---
+# Azure Active Directory UI with ASP.NET Core
+
+The web templates for ASP.NET Core MVC, Razor Pages, and Server-Side Blazor include support for authenticating users with [Azure Active Directory](/azure/active-directory/authentication/overview-authentication) (AAD). Support for AAD is provided by the [Microsoft.AspNetCore.Authentication.AzureAD.UI](https://www.nuget.org/packages/Microsoft.AspNetCore.Authentication.AzureAD.UI) NuGet package. This package contains helper methods and a basic UI that supports authenticating with AAD.
+
+Authentication is enabled with the following highlighted lines in `Startup`:
+
+[!code-csharp[](aad-ui/samples/WebAAD6/Startup.cs?name=snippet&highlight=9,10)]
+
+The preceding code sets up a policy scheme that delegates authentication tasks to an underlying [Open ID Connect](https://openid.net/connect/) authentication handler and sign-in tasks to a cookie authentication handler.
+
+When an unauthenticated user tries to visit a protected page:
+
+* The app emits a challenge to the policy scheme, which is setup as the default scheme.
+* The challenge delegates to the underlying Open ID Connect scheme.
+* Open ID Connect redirects the user to the authorize endpoint.
+
+When the process is successful:
+
+* The app returns to the `signin-oidc` endpoint.
+* The sign-in process continues by validating the response and emitting a cookie with the results.
+
+The authentication package is just a wrapper around the underlying Open ID Connect API. The Open ID Connect API is available to customize any detail or extend the current functionality.
+
+## Customizing the sign-in and sign-out processes
+
+By default the template is setup with a login and logout buttons that call a controller provided in the package to start the authentication operations.
+
+A controller or razor page can be created to begin sign-in and sign-out by calling the appropriate method on the `HttpContext`. For example:
+
+[!code-csharp[](aad-ui/samples/WebAAD6/Areas/Identity/Pages/Account/SignOut.cshtml.cs?name=snippet)]
+
+The preceding sign-out code ensures the app is signed out locally even if remote sign-out fails.
+
+## Configuring the underlying OIDC middleware
+
+This can be achieved by calling Configure after `AddAzureAD` and perfoming any desired customization
+
+```csharp
+services.AddAuthentication(AzureADDefaults.AuthenticationScheme)
```

Comment:
warning CS0618: 'AzureADDefaults' is obsolete: 'This is obsolete and will be removed in a future version. Use Microsoft.Identity.Web instead. See https://aka.ms/ms-identity-web.'